src/vaadin-form-layout.html

```diff
@@ -470,29 +463,34 @@
           }
 
           let col = 0;
-          Array.from(this.children).forEach((child) => {
-            if (getComputedStyle(child).display !== 'none') {
-              let colspan;
+          Array.from(this.children).forEach((child, index) => {
+            if (child.localName === 'br') {
+              // Reset column count on line break
+              col = 0;
+            } else if (getComputedStyle(child).display !== 'none') {
+              const followingVisibleItems = Array.from(this.children)
+                .slice(index + 1)
+                .filter(item => item.localName === 'br' || getComputedStyle(item).display !== 'none');
+
+              const nextLineBreak = followingVisibleItems.length > 0 &&
+                followingVisibleItems[0].localName === 'br';
+
+              if (nextLineBreak) {
+                const colspanRatio = (this._columnCount - col - 1) / this._columnCount;
+                child.style.paddingRight = `calc(${colspanRatio * containerWidth}px + ${colspanRatio} * ${columnSpacing})`;
```

Comment:
Before `paddingRight` was set only on `br` elements, and never on anything else. Now you set both `paddingRight` and `marginRight` on arbitrary user-provided elements at the same time.

Setting them both is unnecessary intrusive. Let’s use only margins.

Reply:
Done. Using margins instead of paddings now.